pkg/operation/botanist/monitoring.go

```diff
@@ -448,6 +434,43 @@ func (b *Botanist) DeleteSeedMonitoring(ctx context.Context) error {
 				Name:      "kube-state-metrics-seed-vpa",
 			},
 		},
+	} {
+		if err := b.K8sSeedClient.Client().Delete(ctx, obj); client.IgnoreNotFound(err) != nil && !meta.IsNoMatchError(err) {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// DeleteSeedMonitoring will delete the monitoring stack from the Seed cluster to avoid phantom alerts
+// during the deletion process. More precisely, the Alertmanager and Prometheus StatefulSets will be
+// deleted.
+func (b *Botanist) DeleteSeedMonitoring(ctx context.Context) error {
+	if err := common.DeleteAlertmanager(ctx, b.K8sSeedClient.Client(), b.Shoot.SeedNamespace); err != nil {
+		return err
+	}
+
+	if err := common.DeleteGrafanaByRole(ctx, b.K8sSeedClient, b.Shoot.SeedNamespace, common.GrafanaOperatorsRole); err != nil {
+		return err
+	}
+
+	if err := common.DeleteGrafanaByRole(ctx, b.K8sSeedClient, b.Shoot.SeedNamespace, common.GrafanaUsersRole); err != nil {
+		return err
+	}
+
+	// TODO: (wyb1) Delete in next minor release
+	if err := b.DeleteKubeStateMetricsSeed(ctx); err != nil {
+		return err
+	}
+
+	for _, obj := range []runtime.Object{
+		&autoscalingv1beta2.VerticalPodAutoscaler{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: b.Shoot.SeedNamespace,
+				Name:      "kube-state-metrics-seed-vpa",
+			},
+		},
```

Comment:
I guess, this one is not needed then, if you call `DeleteKubeStateMetricsSeed` beforehand.